src/sentry/static/sentry/app/views/settings/project/projectIntegrations.jsx

```diff
@@ -0,0 +1,112 @@
+import {Box} from 'grid-emotion';
+import React from 'react';
+
+import {Panel, PanelBody, PanelHeader, PanelItem} from 'app/components/panels';
+import {
+  addIntegrationToProject,
+  removeIntegrationFromProject,
+} from 'app/actionCreators/integrations';
+import {t, tct} from 'app/locale';
+import AsyncComponent from 'app/components/asyncComponent';
+import Button from 'app/components/buttons/button';
+import EmptyMessage from 'app/views/settings/components/emptyMessage';
+import IntegrationItem from 'app/views/organizationIntegrations/integrationItem';
+import Link from 'app/components/link';
+import Switch from 'app/components/switch';
+import TextBlock from 'app/views/settings/components/text/textBlock';
+import space from 'app/styles/space';
+
+export default class ProjectIntegrations extends AsyncComponent {
+  getEndpoints() {
+    let {orgId} = this.props.params;
+    return [['integrations', `/organizations/${orgId}/integrations/`]];
+  }
+
+  onToggleEnabled(enabled, integration) {
+    const {orgId, projectId} = this.props.params;
+
+    const promise = enabled
+      ? addIntegrationToProject(orgId, projectId, integration)
+      : removeIntegrationFromProject(orgId, projectId, integration);
+
+    promise.then(() => {
+      const integrations = [...this.state.integrations];
+      const integRef = integrations.find(i => i.id === integration.id);
+
+      if (enabled) {
+        integRef.projects.push(projectId);
+      } else {
+        integRef.projects = integRef.projects.filter(p => p !== projectId);
+      }
+
+      this.setState({integrations});
+    });
+  }
+
+  renderBody() {
+    const {orgId, projectId} = this.props.params;
+
+    const integrations = this.state.integrations
+      .filter(integration => integration.provider.canAddProject)
+      .map(integration => {
+        const enabled = integration.projects.includes(projectId);
+
+        return (
+          <PanelItem key={integration.id} align="center">
+            <Box flex={1}>
+              <IntegrationItem integration={integration} />
+            </Box>
+            <Box px={2}>
+              <Button
```

Comment:
I can't quite tell, is this view in org or project settings now?

If it's in project settings, we'll want to use `<Feature feature={['project:write']}>` here because I believe users without `project:write` can view project settings, but not edit them.

Reply:
We will want some feature flags around this stuff. Yes this is in the project

Reply:
Yeah let's make sure these controls are only enabled for `project:write` then

Reply:
Also you meant `<Feature access={['project:write']}>`. I'm actually going to go with `project:integrations`, since we have a access scope for this right now since it's slightly different.